client/network/sync/src/lib.rs

```diff
@@ -271,6 +271,8 @@ pub struct ChainSync<B: BlockT, Client> {
 	gap_sync: Option<GapSync<B>>,
 	/// Channel for receiving service commands
 	service_rx: TracingUnboundedReceiver<ToServiceCommand<B>>,
+	/// Handle for communicating with `NetworkService`
+	_network_service: service::network::NetworkServiceHandle,
```

Comment:
Why isn't that used directly?

Reply:
Just trying to find ways to split the work into more reviewable PRs. This PR introduced the provider and next PR which moves the block requests to `ChainSync` will use the `network_service`.